packages/next-on-pages/templates/_worker.js/utils/fetch.ts

```diff
@@ -0,0 +1,40 @@
+export function patchFetchToAllowBundledAssets(): void {
+	const flagSymbol = Symbol.for('next-on-pages bundled assets fetch patch');
+
+	const alreadyPatched = (
+		globalThis.fetch as unknown as { [flagSymbol]: boolean }
+	)[flagSymbol];
+	if (alreadyPatched) {
+		return;
+	}
+
+	applyPatch();
+
+	(globalThis.fetch as unknown as { [flagSymbol]: boolean })[flagSymbol] = true;
+}
+
+function applyPatch() {
+	const originalFetch = globalThis.fetch;
+	globalThis.fetch = async (...args) => {
+		const request = new Request(...args);
+
+		if (request.url.startsWith('blob:')) {
+			try {
+				const url = new URL(request.url);
+				const binaryContent = (
+					await import(`./__next-on-pages-dist__/assets/${url.pathname}.bin`)
+				).default;
+
+				return {
+					arrayBuffer() {
+						return binaryContent;
+					},
+				} as Response;
```

Comment:
Would be nice to expand this to include `.body`, `.text()`, `.json()`, and `.blob()`. But not a big deal.

Reply:
Added 🙂 

I've tested it with the following: https://github.com/dario-piotrowicz/next-apps-for-testing/commit/d7191257b7cb71f57869fa10cb5b0845a32f7e53
(deployed at https://acb42753.next-on-pages-test-5h3.pages.dev/api/inline-assets/)

This is how it compares to `next dev`:
![Screenshot 2023-07-13 at 16 31 29](https://github.com/cloudflare/next-on-pages/assets/61631103/9614f3cb-4d9a-43d4-890c-c7b5b1544535)

It works pretty much the same but we do get different Object types logged, since I create them with the correct constructors I assume that this is a runtime inconsistency (either with `toString` or with the object's creation)

Reply:
Please let me know what you think 🙂

Reply:
PS: yeah I think that the object toString thing is related to the runtime
![Screenshot 2023-07-13 at 16 47 01](https://github.com/cloudflare/next-on-pages/assets/61631103/7747612a-cfab-4ef4-be5c-38157a132694)